### DISCUSSION TITLE
how to known the collection's size and how many rows?

### DISCUSSION THREAD

**Discussion by huangxueliang** [milvus-io/milvus#37648] [Show and tell]
is there a view to get the collections :
how much size it occupy？
how many rows it has?

**Comment by yhmo** [comment]
Row count of a collection:
```
res = collection.query(expr="", consistency_level="Strong", output_fields=["count(*)"])
print(res[0]["count(*)"])
```

No interface to know how much size it occupies. If your milvus is in cluster mode, you can deploy a monitoring system to observe: https://milvus.io/docs/monitor.md

**Reply by huangxueliang** [reply]
is there a monitoring system in Standalone?

**Reply by yhmo** [reply]
> is there a monitoring system in Standalone?

This doc mentioned: "Prometheus then exports metrics of each Milvus component at http://<component-host>:9091/metrics."
For standalone, you can get the metrics from http://localhost:9091/metrics
The returned content is a long list of metrics: https://milvus.io/docs/metrics_dashboard.md
Technically, you can parse the metrics by your self.

**Reply by huangxueliang** [reply]
curl http://localhost:9091/metrics
there is show many metrics,which is the importand metrics ?

**Reply by yhmo** [reply]
All the metrics are described here, usually the latency metrics are important. You can read the doc to pick the metrics that you are interested.
https://milvus.io/docs/metrics_dashboard.md

**Reply by huangxueliang** [reply]
![image](https://github.com/user-attachments/assets/e748465d-4ce9-4444-9d9c-40af39c76d8f)
![image](https://github.com/user-attachments/assets/9ea2ae1e-3f04-4361-b8f5-57a869d9ad20)

**Reply by huangxueliang** [reply]
how can i set the datasource in grafana?

192.168.1.134 is my milvus server(Standalone)

**Reply by huangxueliang** [reply]
![image](https://github.com/user-attachments/assets/826bcd8f-c2dd-43c9-8dd0-a02c7a16339e)
do the Standalone can not use the Prometheus model monitor?
